File: examples/go/runner.go
// Example Gota runner (Go): FNV-1a over the buffer. See ../README.md.
package main

import "gotaexample/gota"

const impl = "go"

func main() {
	gota.Run(impl, func(b *gota.Bencher, data []byte) {
		b.Bench("fnv1a-64", func() {
			var h uint64 = 0xcbf29ce484222325
			for _, x := range data {
				h = (h ^ uint64(x)) * 0x100000001b3
			}
			data[0] ^= byte(h) // sink: consume h
		})
	})
}

File: examples/python/runner.py
#!/usr/bin/env python3
"""Example Gota runner (Python): FNV-1a over the buffer. See ../README.md."""

from __future__ import annotations

import gota

IMPL = "python"
MASK = 0xFFFFFFFFFFFFFFFF
PRIME = 0x100000001B3


def register(b: gota.Bencher, data: bytearray) -> None:
    def fnv1a():
        h = 0xCBF29CE484222325
        for x in data:
            h = ((h ^ x) * PRIME) & MASK
        data[0] ^= h & 0xFF  # sink: consume h so the loop can't be optimized away

    b.bench("fnv1a-64", fnv1a)


if __name__ == "__main__":
    gota.run(IMPL, register)
